dandi/dandiapi.py

```diff
@@ -885,6 +891,9 @@ def __init__(self, **data: Any) -> None:
         # underscores, so we have to do it ourselves.
         self._metadata = data.get("metadata", data.get("_metadata"))
 
+    def __str__(self) -> str:
+        return self.identifier
```

Comment:
if we have access to instance name via `.client`, please make it `{instance_name.upper()}:assets/{self.identifier}`

Reply:
@yarikoptic The client only stores the API URL.  While `known_instances_rev` can be used to look up a corresponding instance name, the URL could be anything, not just a known URL.

Reply:
If unknown url, just use entire url for now and the name. At least this way we/user would know exactly about what that thing is from

Reply:
For all pragmatic use cases I expect us to get a nice and meaningful name

Reply:
Done.